scripts/surface_high_engagement_issues.py

```diff
@@ -0,0 +1,407 @@
+    response = httpx.post(
+        url, headers=headers, json={"query": query, "variables": variables}
+    )
+    response.raise_for_status()
+    data = response.json()
+    try:
+        item_id = data["data"]["addProjectV2ItemById"]["item"]["id"]
+        print(f"Added issue to project with item ID: {item_id}")
+    except Exception:
+        print(f"Failed to add issue to project: {data}")
+    return item_id
+
+
+def add_needs_attention_label(issue_number: int, headers: dict):
+    """
+    Add a "needs:attention" label to the issue.
+
+    Args:
+        issue_number (int): The number of the issue.
+        headers (dict): HTTP headers for GitHub API requests.
+    """
+    url = f"https://api.github.com/repos/{GITHUB_REPO}/issues/{issue_number}/labels"
+    response = httpx.post(url, headers=headers, json=["needs:attention"])
+    response.raise_for_status()
+    print(f"Added 'needs:attention' label to issue #{issue_number}")
+
+
+def set_needs_priority_status_on_high_engagement_issues(new_comment_interval_days: int):
+    """
+    Sets the "Needs Priority" status on high engagement issues with new comments.
+    This status is a field on the Prefect Backlog project board.
+
+    Args:
+        new_comment_interval_days (int): Interval in days to check for new comments.
+
+    Example:
+        python surface_high_engagement_issues.py --new_comment_interval_days 1
+
+        This will check for high engagement issues with new comments within the last day and add the "Needs Priority" status to them.
+        You may want to ad-hoc run this for a longer interval. For instance, if you wanted to loop through high-engagement issues that
+        have had new comments in the last year, you could run:
+
+        python surface_high_engagement_issues.py --new_comment_interval_days 365
+    """
+    token = get_github_token()
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "Accept": "application/vnd.github.v3+json",
+    }
+
+    high_engagement_issues = get_high_engagement_issues(headers)
+    for issue in high_engagement_issues:
+        issue_number = issue["number"]
+        if issue_has_new_comment(
+            issue, new_comment_interval_days, headers
+        ) or not prioritized_recently(issue_number, headers):
+            # Issue id is a globally unique identifier
+            issue_id = get_issue_id(issue_number, headers)
+            project_item_id = get_project_item_id(issue_id, headers)
+            if project_item_id is None:
+                print(f"Issue #{issue_number} is not in the project. Adding it now.")
+                project_item_id = add_issue_to_project(issue_id, PROJECT_ID, headers)
+            else:
+                print(f"Issue #{issue_number} is already in the project.")
+                add_needs_attention_label(issue_number, headers)
+                print(
+                    f'Added "needs:attention" label to issue #{issue_number} in the project.'
+                )
```

Comment:
@zzstoatzz I changed the implementation slightly due to this [convo](https://prefecthq.slack.com/archives/C0796KB678S/p1719570335437389). So now we just add a `needs:attention` label if it's in the project, instead of updating its status. This means I don't need the functions to retrieve the field id or option ids, and I don't need the `update_project_status` function either. I did add a new `add_needs_attention_label` function though.